### PR TITLE
Post to Flowdock integration platform

### PR DESCRIPTION
**Pull request by RedBulli**
Post to Flowdock integration platform which then constructs the activities for Flowdock. Supports multiple tokens (comma separated)

### CODE REVIEW

File: src/main/java/com/flowdock/plugins/confluence/FlowdockConnection.java [hunk 1]
```diff
@@ -4,71 +4,50 @@
...
+		out.write(params.serialize());
 		out.close();
-		
+
 		BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
 		while (in.readLine() != null);
 		in.close();
-		
+
 		conn.connect();
```
Comment (aesterline) [inline src/main/java/com/flowdock/plugins/confluence/FlowdockConnection.java]: I wonder if this is really needed. It seems like we have already written/read to/from the connection. The [docs](http://docs.oracle.com/javase/7/docs/api/java/net/URLConnection.html#connect()) seem to indicate that conn.connect() would be ignored if it was already opened. We might want to move this line up to right after conn.setRequestProperty.

File: src/main/java/com/flowdock/plugins/confluence/FlowdockConnection.java [hunk 2]
```diff
@@ -4,71 +4,50 @@
...
 		conn.setDoInput(true);
+		conn.setRequestProperty("Content-Type", "application/json");
 		conn.setRequestProperty("User-Agent", "Flowdock Confluence Plugin");
-		
+
 		OutputStreamWriter out = new OutputStreamWriter(conn.getOutputStream());
-		String postData = mapToPostData(params);
-		out.write(postData);
+		out.write(params.serialize());
 		out.close();
```
Comment (aesterline) [inline src/main/java/com/flowdock/plugins/confluence/FlowdockConnection.java]: A better pattern for closing out the streams would be something like...

```java
OutputStreamWriter out = new OutputStreamWriter(conn.getOutputStream());
try {
  out.write(params.serialize());
} finally {
  out.close();
}
```

File: src/main/java/com/flowdock/plugins/confluence/FlowdockEventRenderer.java [hunk 3]
```diff
@@ -25,71 +45,126 @@
...
+		if (comment.getParent() != null)
+			result.setProperty("parent", renderObjectData(comment.getParent()));
+		result.setProperty("container", renderContentJson(comment.getContainer()));
+		return result;
+	}
+
+	private JsonObject renderPageData(Page page) {
+		JsonObject result = renderObjectData(page);
+		JsonArray ancestors = new JsonArray();
+		for (Iterator<Page> i = page.getAncestors().iterator(); i.hasNext();) {
```
Comment (aesterline) [inline src/main/java/com/flowdock/plugins/confluence/FlowdockEventRenderer.java]: You should be able to use an [enhanced for loop](http://www.javacertifications.net/javacert/enhancedforloop.jsp) here. Something like..

```java
for(Page ancestor : page.getAncestors) {
}
```

File: src/main/java/com/flowdock/plugins/confluence/FlowdockEventRenderer.java [hunk 4]
```diff
@@ -25,71 +45,126 @@
...
+	private JsonObject renderObjectData(ContentEntityObject object) {
+		JsonObject result = new JsonObject();
+		result.setProperty("title", object.getTitle());
+		result.setProperty("content", object.getBodyAsStringWithoutMarkup());
+		result.setProperty("html_content", renderAsHtml(object));
+		result.setProperty("url", getBaseUrl() + object.getUrlPath());
+		result.setProperty("id", String.valueOf(object.getId()));
+		result.setProperty("type", object.getType());
+		JsonArray arr = new JsonArray();
+		for (Iterator<Label> i = object.getLabels().iterator(); i.hasNext();)
```
Comment (aesterline) [inline src/main/java/com/flowdock/plugins/confluence/FlowdockEventRenderer.java]: Same as above. Use the enhanced for loop.

File: src/main/java/com/flowdock/plugins/confluence/FlowdockConnection.java (hunk 2 above)
Comment (sakari) [inline src/main/java/com/flowdock/plugins/confluence/FlowdockConnection.java]: what java we are targeting? I was looking at whether `.close` would be needed at all and came across `try`-with-resources in java 7

Comment (aesterline) [inline src/main/java/com/flowdock/plugins/confluence/FlowdockConnection.java]: It looks like we are targeting 1.6. You can find it declared in the pom.xml.

File: src/main/java/com/flowdock/plugins/confluence/FlowdockConnection.java (hunk 1 above)
Comment (RedBulli) [inline src/main/java/com/flowdock/plugins/confluence/FlowdockConnection.java]: It does indeed work without the connect().

**Comment by sakari** [comment]
seems legit, @aesterline do you want to do something with this still?

**Comment by aesterline** [comment]
No. I think it seems good to merge.

**Pull request merged**